llvm/lib/IR/RepoTicket.cpp

```diff
@@ -77,76 +78,55 @@ const Constant *getAliasee(const GlobalAlias *GA) {
   return Target;
 }
 
-static const DependenciesType &
-updateInitialDigestAndGetDependencies(const GlobalObject *GO, MD5 &GOHash,
-                                      GOInfoMap &GOIMap, GONumber &GONum) {
-  GOInfoMap::const_iterator Pos;
+static const GODigestState
+updateInitialDigestAndGetGOStatus(const GlobalObject *GO, MD5 &GOHash,
+                                  GOInfoMap &GOIMap) {
+
   if (const auto GV = dyn_cast<GlobalVariable>(GO)) {
-    ++GONum.VarNum;
-    Pos = calculateInitialDigestAndDependenciesAndContributions(GV, GOIMap);
+    calculateInitialDigestAndDependenciesAndContributions(GV, GOIMap);
   } else if (const auto Fn = dyn_cast<Function>(GO)) {
-    ++GONum.FuncNum;
-    Pos = calculateInitialDigestAndDependenciesAndContributions(Fn, GOIMap);
+    calculateInitialDigestAndDependenciesAndContributions(Fn, GOIMap);
   } else {
     llvm_unreachable("Unknown global object type!");
   }
-  GOHash.update(Pos->second.InitialDigest.Bytes);
-  return Pos->second.Dependencies;
+  const GOInfo &GOInformation = GOIMap[GO];
+  GOHash.update(GOInformation.InitialDigest.Bytes);
+  return std::forward_as_tuple(GOInformation.Contributions,
+                               GOInformation.Dependencies, true);
 }
 
-GODigestState updateDigestGONumAndGetDependencies(const GlobalObject *GO,
-                                                  MD5 &GOHash,
-                                                  GOInfoMap &GOIMap,
-                                                  GONumber &GONum) {
-  auto It = GOIMap.find(GO);
-  if (It != GOIMap.end()) {
-    const GOInfo &GOInformation = It->second;
-    GOHash.update(GOInformation.InitialDigest.Bytes);
-    return {GOInformation.Dependencies, true};
-  }
-
-  return {updateInitialDigestAndGetDependencies(GO, GOHash, GOIMap, GONum),
-          true};
+static const GODigestState updateDigestAndGetGOState(const GlobalObject *GO,
+                                                     MD5 &GOHash,
+                                                     GOInfoMap &GOIMap) {
+  const GOInfo &GOInformation = GOIMap[GO];
+  GOHash.update(GOInformation.InitialDigest.Bytes);
+  return std::forward_as_tuple(GOInformation.Contributions,
+                               GOInformation.Dependencies, true);
```

Comment:
This seems to appear in 3 places. Could it be factored out?

Reply:
Refactored the code, thanks.